### PR TITLE
Preps for v2

### PR DESCRIPTION
**Pull request by TonyB9000**
You are cordially invited to review this updates.  The update to util.py "find_mpas_files" will now match both case-named (v2) and non-case-named (v1) data files.  The MPAS cmor handlers (and mpas.py "remap" function) now pass in "mpasocean" or "mpasseaice" pcode parameter, allowing remap to select "-P pcode" appropriately for ncremap call.  I hope this won;t cause problems for v1 datasets, or if people run "pre -P pcode" versions of ncremap.

Until such changes as these are released, I cannot update the DataSM to run MPAS CMIP6 postprocessing.

### CODE REVIEW

**Comment by xylar** [comment]
@TonyB9000, could you let me know what you're looking for in a review? Do you have example output you want me to look at?  Do you just want me to do a visual inspection of the code?

**Comment by xylar** [comment]
It might be good to have @czender take a look at the ncremap calls (particularly if you can provide the actual commands being used for a few variables, as opposed to just the code).

File: e3sm_to_cmip/util.py
Comment (xylar) [inline e3sm_to_cmip/util.py:876]: @TonyB9000, could you add a comment here explaining this second term?  I believe we now prepend the name of the simulation.  It the issue that the name of the simulation contains dots?  If so, I wonder how robust this will be.

Comment (xylar) [reply e3sm_to_cmip/util.py:876]: I think the more robust solution would be to add a `.*` or something at the beginning of the pattern.  I'm not super good with regular expression but there's definitely a pattern like `.*` for any number of characters.

Comment (TonyB9000) [reply e3sm_to_cmip/util.py:876]: I was looking into that, thanks.  This was quick and dirty, and assumes the "case-name" prefix always has 3 components, which is not terribly robust.  I'll conduct a test for eliding initial characters.

Comment (TonyB9000) [reply e3sm_to_cmip/util.py:876]: Regarding the ncremap calls (in mpas.py), I'm not sure how best to produce a representative sample of command lines.

The additions of "-P <pcode>" are required by Zender, unless the mapfile name contains the string "sgs".  I don't know if this is a no-op for v1 usage, so I apply it everywhere.  I had to instrument the cmor handlers for mpaso and mpassi variables in order to issue the switches for "-P mpasocean" and "-P mpasseaice", etc.

Comment (czender) [reply e3sm_to_cmip/util.py:876]: Maps with names `map*sgs*.nc` should no longer be used. Period. Use the `-P model_name` formulation instead, and replace any map-file names of the form `map*sgs*.nc` with the corresponding map-file that has no `sgs` in the name. The resulting commands will work on both v1 and v2 files.

**Review by czender — COMMENTED** [review]
Be aware that -P mpasseaice requires access to the variable timeMonthly_avg_iceAreaCell for the same time period as the variable(s) being regridded. The variable can be stored in the same file or in a different file than the variables being regridded. In the latter case it is necessary to point ncremap to the external file with the --sgs_frc option, as described in the documentation.

**Comment by TonyB9000** [comment]
@czender I expect our only CMIP6 mpassi generation (any variables) will derive from the native "sea-ice monthly" raw model-output, wherein the timeMonthly_avg_iceAreaCell variable will be present along with all others.

I will search for the non-"sgs" files corresponding to those we had formerly used.  Thanks!

**Comment by TonyB9000** [comment]
@chengzhuzhang I do not know how to recreate or correct the "failed" test listed above, whose error reads:
{'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}*

File: e3sm_to_cmip/util.py
Comment (TonyB9000) [reply e3sm_to_cmip/util.py]: @czender Thanks! Especially good to know I don't need to switch on v1 vs v2 for this.

An issue with the e3sm_to_cmip repo is that the "scripts/cwl_workflows/*" contain <job>.yaml scripts, providing input parameters that are out of date and often reference people's personal directories.  I will attempt to clean that material as an adjunct to ensuring "-P model_name" is applied where appropriate (and eliminating use of "*sgs*" maps.)

**Comment by chengzhuzhang** [comment]
@TonyB9000 I did a rebase to xylar's branch . So you  and I should be able to update the local branch and work on testing. [I'm not too familiar with this operation either, hope it works]

**Comment by TonyB9000** [comment]
So, I should "git rebase separate_ocean_and_seaice_masks"

I get:  "fatal: invalid upstream 'separate_ocean_and_seaice_masks'"

Then pip install/test?  Something?

**Comment by TonyB9000** [comment]
I'd like to "squash and merge", then delete my branch, and then do a pull of master.

Does that sound ok?

**Comment by chengzhuzhang** [comment]
> I'd like to "squash and merge", then delete my branch, and then do a pull of master.
> 
> Does that sound ok?

No. Could you try 
git pull origin preps_for_v2

**Comment by TonyB9000** [comment]
OK.  That popped up a "Please enter a commit message to explain why this merge is necessary".  So, I'll add a comment and save.

**Comment by TonyB9000** [comment]
Now, status says:

On branch preps_for_v2
Your branch is ahead of 'origin/preps_for_v2' by 9 commits.
(use "git push" to publish your local commits)

nothing to commit, working tree clean

Should I do another push?

(I added the comment on Xylar'r branch.  So confusing ...)

**Comment by TonyB9000** [comment]
I'm going to "pip install" what I now have to the e2c_dev environment and run tests.

**Comment by chengzhuzhang** [comment]
> I'm going to "pip install" what I now have to the e2c_dev environment and run tests.

Did you get all the code changes? In this case, we won't need @tomvothecoder to look at it.

**Comment by TonyB9000** [comment]
Yes, I got all the changes, then did a local install and I'm retesting (ocean at least - can't test sea-ice until Zender reports back)

**Comment by TonyB9000** [comment]
It still said that my branch "prep_for_v2" is nine commits ahead of remote, or something.  But locally I have all the changes.

**Comment by tomvothecoder** [comment]
> It still said that my branch "prep_for_v2" is nine commits ahead of remote, or something. But locally I have all the changes.

This is because Jill rebased "prep_for_v2" on top of Xylar's branch. The quick way to get rid of this message is to stash any additional changes, delete your local copy of "prep_for_v2", and the checkout the latest "prep_for_v2" (then pop changes if needed).

Also, I noticed the GH Actions codecov report is lower now with the code changes. You can ignore this.

**Comment by TonyB9000** [comment]
You realize when you say "git stash, delete local branch, and checkout latest", my impulse is to "git stash, rm -rf repo, git clone e3sm_to_cmip, and then see what remains.  I'm guessing that might destroy the git stash.

Are you saying, after "git stash", I should "git checkout master"? (because git won't let me delete an active branch).
Then, after "git -d preps_for_v2", how do I "checkout latest preps_for_v2"?   I need commands - hints confuse me :(

**Comment by tomvothecoder** [comment]
> You realize when you say "git stash, delete local branch, and checkout latest", my impulse is to "git stash, rm -rf repo, git clone e3sm_to_cmip, and then see what remains. I'm guessing that might destroy the git stash.
> 
> Are you saying, after "git stash", I should "git checkout master"? (because git won't let me delete an active branch). Then, after "git branch -d preps_for_v2", how do I "checkout latest preps_for_v2"? I need commands - hints confuse me :(

Hey @TonyB9000, I recommend looking on StackOverflow or other resources for things like Git and trying your best at tackling these issues first. I understand that there are many terms to Git that can be confusing, but you only need to remember a handful of commands for your workflow for most cases.

Example: https://stackoverflow.com/questions/315911/git-for-beginners-the-definitive-practical-guide

My comment was merely a high-level guide rather than a step-by-step to solve the issue, which might result in some vagueness in the details. It can be a time sink to debug Git over email or GitHub comments, especially since I know we're all juggling multiple priorities.

For more specificity in this case:
* `git stash` -- stash your local changes of "preps_for_v2"
* `git checkout master` -- get off of "preps_for_v2" branch
* `git branch -D preps_for_v2 -- delete your local copy of the branch
* `git branch checkout preps_for_v2` -- get the remote copy of the branch (latest version)
* `git stash pop` - pop your changes on top of the branch

**Comment by chengzhuzhang** [comment]
Thank you @tomvothecoder . After rebasing, I suggested `git pull origin prep_for_v2` to Tony, without realizing the additional steps Tony has to do on his local. 

@TonyB9000 I strongly recommend that you follow what Tom suggested. Any codes that have already pushed to remote, they won't get missing.  So don't be afraid of trying different things suggested by Stackoverflow..

**Comment by TonyB9000** [comment]
Thank You both, @tomvothecoder  and @chengzhuzhang .  I will do that.

(I've never tried a "git branch checkout" of a branch I've deleted - new to me.  Git ramps up my paranoia...)

**Comment by TonyB9000** [comment]
"git stash" reported "no local changes to save" (having already pushed them earlier.)
After "git checkout master" and "git branch -D preps_for_v2",  no complaints.

Then "git branch checkout preps_for_v2" says "fatal: Not a valid object name: 'preps_for_v2'. (it must have thought "checkout" was a branch, and "preps" was a file...)

But (yes, thanks to the Interwebs), "git checkout preps_for_v2" worked ok.

Both Xylar's and my own changes are present.  Thanks!

**Comment by TonyB9000** [comment]
Note:  For the contemporary ncremap call (in mpas.py) I eliminated the redundant or unnecessary flags
--no_stg_grd, --msk_src, and --mask_dst (per Charlie Zender's recommendation)

However, for the case where "mapfile contains sgs" I retained the conditional command line.  Perhaps it should be removed entirely.

**Comment by TonyB9000** [comment]
In keeping with Charlie's recommendation, both the v1 and v2 land processing employ "-P elm, -m mapfile" and no longer "-s src_grid -d dst_grid -sgs_frc= ..." methodology.  Tested both v1 and v2 land data, min tsl values ~231 Kelvin.

Now, we could employ "lnd-elm" workflow everywhere, by passing the file match parameter "clm2" or "elm" as needed.

File: scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl
Comment (chengzhuzhang) [inline scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Quotes on elm is not needed. I think we should add other flags, similar to eam:
  - -7
  - --dfl_lvl=1
  - --no_cll_msr
  - --no_stdin

**Review by chengzhuzhang — CHANGES_REQUESTED** [review]
I think this PR can be merged after review. Another PR focusing on sea-ice will be reviews later. 

We need some additional flags in ncremap calls for CMIP compliance. Would you please update and test?
I'm wondering, what is the difference between the workflows: lnd-n2n and lnd-elm?

File: scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl
Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: I can update and test.  As far as "n2n" and "elm", I note that our v1 work (in datasm, as written by Sterling) called "n2n", and v2 called "elm".  BOTH were written with the "-s -g -sgs_frc" method, and having converted both to "-P elm, -m mapfile", the only remaining difference is that n2n searches for files with "clm2", while elm uses "elm".  If we can pass "search_patt: clm2" or "search_patt: elm" as a parameter to the workflow - both become identical.

Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Well, this is unfortunate:  I added below "ncremap"

  - ncremap
  - -7
  - --dfl_lvl=1
  - --no_cll_msr
  - --no_stdin

But got never-before-seen errors:

ERROR Tool definition failed initialization:
Tool definition file:///p/user_pub/e3sm/staging/resource/cmor/cwl_workflows/lnd-elm/ncremap_lnd.cwl failed validation:
../../../../../../staging/resource/cmor/cwl_workflows/lnd-elm/ncremap_lnd.cwl:2:1:     Object
                                                                                      `../../../../../../staging/resource/cmor/cwl_workflows/lnd-elm/ncremap_lnd.cwl`
                                                                                      is not valid because
                                                                                        tried
                                                                                        `CommandLineTool`
                                                                                        but
../../../../../../staging/resource/cmor/cwl_workflows/lnd-elm/ncremap_lnd.cwl:8:1:         the `arguments`
                                                                                          field is not valid
                                                                                          because
                                                                                            tried array
                                                                                            of <string or
                                                                                            Expression or
                                                                                            CommandLineBinding>
                                                                                            but
../../../../../../staging/resource/cmor/cwl_workflows/lnd-elm/ncremap_lnd.cwl:16:5:           item is
                                                                                              invalid because
                                                                                                - tried
                                                                                                string but
                                                                                                  the
                                                                                                  value is not string
                                                                                                - tried
                                                                                                Expression but
                                                                                                  value
                                                                                                  is a int but
                                                                                                  expected a string

Comment (chengzhuzhang) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: I suspect it could be a syntax error? maybe following this example. https://github.com/E3SM-Project/e3sm_to_cmip/blob/61d7fdf4d17f8ce814b9bc786d1c1c089221968a/scripts/cwl_workflows/atm-unified-eam/hrzremap_posin.cwl

Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Regarding
  - ncclimo
  - '-7'
  - --dfl_lvl=1
  - --no_cll_msr
  - --no_stdin
  - -P
  - "eam"
  
  As you suggested, the quotes on '"eam"' are not needed (works fine either way.)
  Single-quoting (maybe double as well) the '- 7' worked.  But including  the "--no_cll_msr" led to the error from ncap2
  
  ncap2: ERROR out(): Could not read var area
  and 
  ncremap: ERROR Failed post-processing to convert output area from [sr] to [km2]. Debug this:
ncap2 -O -s 'area*=6.37122e6^2/1.0e6;area@long_name="Gridcell area";area@units="km^2"' "/p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmp_u2v6xgd/v2.LR.abrupt-4xCO2_0101.e
lm.h0.0001-01.nc" "/p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmp_u2v6xgd/v2.LR.abrupt-4xCO2_0101.elm.h0.0001-01.nc"

I'll skip that one, and see if it complains about "--no_stdin"

Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: And of course it does.  Changing

```
  - -P
  - elm
  - -m
  - $(inputs.remapfile)
  - -v
  - $(inputs.var_list.join(','))

inputs:
  lnd_files:
    type: File
  account:
    type: string
  partition:
    type: string
  timeout:
    type: string
  remapfile:
    type: string
  var_list:
    type: string[]

stdin: $(inputs.lnd_files.path)
```

to
```

  - -P
  - elm
  - -m
  - $(inputs.remapfile)
  - -v
  - $(inputs.var_list.join(','))
  - $(inputs.lnd_files.path)

inputs:
  lnd_files:
    type: File
  account:
    type: string
  partition:
    type: string
  timeout:
    type: string
  remapfile:
    type: string
  var_list:
    type: string[]
```

Produced the obscure error

    -P \
    elm \
    -m \
    /p/user_pub/e3sm/staging/resource/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc \
    -v \
    TSOI \
    /p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmpqiwbh3fy/stgcce0aaff-ebe8-4532-96a1-1e2c56555781/lnd_files.txt
Input #00: /p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmpqiwbh3fy/stgcce0aaff-ebe8-4532-96a1-1e2c56555781/lnd_files.txt
Map/Wgt  : /p/user_pub/e3sm/staging/resource/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
nco_err_exit(): ERROR Short NCO-generated message (usually name of function that triggered error): nco__open()
nco_err_exit(): ERROR Error code is -51. Translation into English with nc_strerror(-51) is "NetCDF: Unknown file format"
ERROR: nco__open() unable to open file "/p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmpqiwbh3fy/stgcce0aaff-ebe8-4532-96a1-1e2c56555781/lnd_files.txt"
ERROR NC_ENOTNC Not a netCDF file

And that file disappears as soon as it is created:

(e2c_dev2) ll tmpqiwbh3fy/stgcce0aaff-ebe8-4532-96a1-1e2c56555781/lnd_files.txt
ls: cannot access tmpqiwbh3fy/stgcce0aaff-ebe8-4532-96a1-1e2c56555781/lnd_files.txt: No such file or directory

BTW, the example you gave is for ncclimo, not ncremap, if that matters.

Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Also, curiously, the ncclimo variables input (example) says

 - $(inputs.variables.join(" "))        [space-separated]
 
 whereas the variables input supplied to ncremap are comma-separated.
 
   - $(inputs.var_list.join(','))

Comment (chengzhuzhang) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Tony, are you saying that neither `--no_cll_msr` nor `--no_stdin` works?

Comment (TonyB9000) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Jill, the `--no_cll_msr` caused a `ncap2: ERROR out(): Could not read var area`.

The `--no_stdin` might work, but it will force you to revamp the workflow in ways in which I am unfamiliar.  You certainly cannot just add it to the command line.  The existing workflow "lnd.cwl" has subordinate workflows "discover_lnd_files.cwl", whose output is input to "ncremap_lnd.cwl", whose output is input to ""timeseries.cwl"", whose output is input to "cmor.cwl".  Each of these has a peculiar way of specifying input and output.

The existing "ncremap_lnd.cwl" has an ncremap command line that terminates with the "var list".  But when it is run, the logfile shows it ends with

```
-v \
    TSOI < /p/user_pub/e3sm/bartoletti1/Pub_Work/2_PostProcess/altprocess/ManualTest/ncremap/tmp0fqntji7/stg3c63d5e6-2b79-45d6-9917-e4c0542af9ee/lnd_files.txt
```
and this works fine (don't ask me how.  I assume "lnd_files.txt" is a listing of the files.)

The script has the line:

`stdin: $(inputs.lnd_files.path)`
    
So I assumed I could add the "--no_stdin" and just add "$(inputs.lnd_files.path)" to the command line.  But that does not work.  If CWL supported print statements I could inspect some of these terms, but it does not.  I suppose I could create a "print.cwl" workflow, and call it as a separate step, giving it something I want to see printed - ridiculous...

**Comment by TonyB9000** [comment]
This is as close as I could get and have a successful test.
```

arguments:
  - -A
  - $(inputs.account)
  - --partition
  - $(inputs.partition)
  - -t
  - $(inputs.timeout)
  - ncremap
  - '-7'
  - --dfl_lvl=1
  - -P
  - elm
  - -m
  - $(inputs.remapfile)
  - -v
  - $(inputs.var_list.join(','))

stdin: $(inputs.lnd_files.path)
```

File: scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl
Comment (chengzhuzhang) [reply scripts/cwl_workflows/lnd-elm/ncremap_lnd.cwl:17]: Thanks for clarifying. I was not aware that the `eam` workflow calls `ncclimo`, while the `elm` workflow calls `ncremp`. I thought both can have a same ncclimo/ncremp call. I think you can push the code change (leaving out --no_cll_msr and --no_stdin. And I will tag Charlie for a final pass. Thank you for experimenting.

**Review by chengzhuzhang — CHANGES_REQUESTED** [review]

**Comment by TonyB9000** [comment]
Hey Jill,

Actually, after calling ncremap, the output of ncremap_lnd.cwl becomes the input to "time_series_lnd.cwl", which DOES call ncclimo.  I have not inspected that command line yet.

**Comment by czender** [comment]
The v1 workflow for ELM is "inverted" relative to EAM. EAM does ncclimo then ncremap and ELM did ncremap then ncclimo. That was due to limitations in the v1 regridder, now solved. For v2, ELM can (and I think should) call the same workflow as EAM, i.e., ncclimo then ncremap, with the appropriate -P options. And for that matter MPAS-Ocean could follow the same formula. As you know, MPAS-Seaice is a special case because the of the time-varying SGS field. I fixed the --no_cll_msr issue with ELM in the latest snapshot, which I'm testing now. --rnr_thr also works as expected with SGS variables now. Will try to release in --alpha13 for your testing by tomorrow.

**Comment by TonyB9000** [comment]
Thanks for that clarification, Charlie.  When you say "For v2, ELM can ... call the same workflow as EAM", I am hoping we can do the same for v1 data, for which there are many deferred CMIP6 land publications to generate.  I am striving to avoid having to bifurcate our codes to accommodate v1/v2/(v3?), etc.  Selecting a v1 mapfile, versus a v2 mapfile is trivial, but having to support nearly-duplicated workflows with minor variations is less than ideal...

**Comment by czender** [comment]
Fear not, it's all backwards compatible, v2 workflows will work on v1 data.

**Comment by czender** [comment]
Please try again with NCO 5.1.1-alpha13, just released. This should fix the problems with `--no_cll_msr` and `--rnr_thr`.

**Comment by TonyB9000** [comment]
I am conducting a tests with NCO 5.1.1alpha13, with v1 and v2 land data (variable = tsl).
In the process, I have also modified the cwl-workflow "lnd_elm" to accept a "file_pattern" parameter (".clm2.h0", or ".elm.h0"), respectively for v1 or v2 data.  If this works, there will no longer be a need to support two separate cwl workflows (we can dispense with "lnd-n2n" and its subordinate workflows, whose only differences were the file-patterns employed).

**Comment by TonyB9000** [comment]
Both v1 and v2 tests are successful with NCO 5.1.1alpha13.  Min values for tsl were
tsl, size = 1 NC_FLOAT, value = 231.397 (for v20221024/tsl_Lmon_E3SM-1-0_1pctCO2_r1i1p1f1_gr_000101-005012.nc)
tsl, size = 1 NC_FLOAT, value = 224.167 (for v20221024/tsl_Lmon_E3SM-2-0_1pctCO2_r1i1p1f1_gr_000101-005012.nc)

**Comment by chengzhuzhang** [comment]
@TonyB9000 this is good news. Can you please push the code changes, finalized ncremap call and with "lnd-n2n" dropped.

**Comment by TonyB9000** [comment]
Done.  All e2c changes pushed.

**Comment by czender** [comment]
@TonyB9000 a little known feature: Calling `ncclimo` with `-P clm` (instead of `-P elm`) causes it to use the string `clm2` (instead of `elm`) when looking for input filenames. Everything else will behave identically to `-P elm`, including in `ncremap`. Feel free to take advantage of that feature if it simplifies your scripts by eliminating the `-m mdl_nm` option. That's what it's there fore :)

**Comment by TonyB9000** [comment]
Thanks for the tip, Charlie.  Hopefully I can further reduce to constellation of codes used to generate CMIP6 for land (and others).  Presently, the "main" (lnd.cwl) calls upon 7 subordinate clw workflow files for its various "steps".  In order of execution:

```
lnd.cwl
find_casename.cwl            [uses the ".clm2.h0" or ".elm.h0" pattern]
get_start_end.cwl
generate_segments.cwl
discover_lnd_files.cwl        [uses the ".clm2.h0" or ".elm.h0" pattern]
ncremap_lnd.cwl              [calls ncremap]
time_series_lnd.cwl          [calls ncclimo]
cmor.cwl                         [calls e3sm_to_cmip]
```

I am not a cwl_workflow coder - but I play one on TV.   In the "time_series" step, the call to ncclimo still includes "-P elm" and "--no_cll_msr", but has never had "-m mdl_nm" applied (as far as I can tell).  It appeared to work fine as is with both v1 (clm2) and v2 (elm) file data...

**Comment by czender** [comment]
Never mind, then. Now that I think about it more carefully, this feature is currently only useful for generating climos, not timeseries.

**Comment by TonyB9000** [comment]
No Problem.  I will see about how we handle climo generation as well.  Thanks.

**Review by chengzhuzhang — APPROVED** [review]
LGTM. Thank you for the PR.

**Comment by TonyB9000** [comment]
@chengzhuzhang @tomvothecoder  I will go ahead with the merge - but note I do not know how to do a release, and until that happens the changes will be unavailable for testing in datasm.

**Comment by tomvothecoder** [comment]
@chengzhuzhang @TonyB9000 Should we wait until #155 is merged first until we perform a release, since #155 looks like a continuation of this PR?

**Comment by chengzhuzhang** [comment]
Yes, that's my preference too. Regarding to the other PR, Xylar is actively reviewing it. And hopefully we can merge soon.

**Comment by chengzhuzhang** [comment]
@tomvothecoder I think we can pull the trigger for a new release, v1.9 maybe?

**Comment by tomvothecoder** [comment]
@chengzhuzhang and @TonyB9000, v1.9.0 is now available on `conda-forge`.

https://anaconda.org/conda-forge/e3sm_to_cmip

**Comment by TonyB9000** [comment]
Excellent!  I will first update my "e2c_dev" (conda-envs/dev.yml) to reflect the latest e2c.  Then, rebuilding my conda "e2c_dev" for my stand-alone e2c testing.  (Should I pull e2c source down anew?  Perhaps wipe and just git clone?)

Then I will pip install the latest datasm codes (into dsm_dev) for testing the changes to the jobs-preparation codes (to address the changes made to the e2c cwl_workflows).  Hoping not to see too many surprises.  When clean, I will PR the datasm changes.

**Pull request merged**
